src/reporter/index.js

```diff
@@ -43,7 +44,8 @@ export default class Reporter {
             screenshotPath: reportItem.screenshotPath,
             screenshots:    reportItem.screenshots,
             quarantine:     reportItem.quarantine,
-            skipped:        reportItem.test.skip
+            skipped:        reportItem.test.skip,
+            contexts:       sortBy(reportItem.contexts, 'userAgent')
```

Comment:
I guess sorting is not necessary.
Custom reporter plugin should decide itself will it sort contexts or not.